### DISCUSSION TITLE
Exporting to star produced no output

### DISCUSSION THREAD

**Discussion by JuhaHuiskonen** [nextpyp/.github#75] [Q&A] [answered]
Hello,  I tried to use Particle filtering block to export metadata to STAR.   However, nothing was written in the output folder.  No errors were printed either and a new reconstruction using the new set of particles was calculated ok. What should I try next? 

<img width="1503" alt="Screenshot 2024-02-23 at 8 29 04 1" src="https://github.com/nextpyp/.github/assets/20352352/6a7b3542-bf54-42f4-a72f-36b1145177d7">

**Comment by yezhou867** [comment]
By default, the converted files go to `current_project/relion/ `folder, we probably didn't really take the output parameter. Will try to fix it.

**Comment by JuhaHuiskonen** [comment]
I rerun the particle filtering block and this time I removed the `Output folder` parameter.  However, there was no `current_project/relion/` created.

**Reply by yezhou867** [reply]
Sorry, the `current_project` may be a little misleading.  I mean the project path accessed by `show filesystem location` from the block.

**Reply by JuhaHuiskonen** [reply]
I checked the project folder and the folder of the particle filter block.  There is no `relion` folder.

**Reply by yezhou867** [reply, marked as answer]
Oh, I missed the part that you were using a particle filter job to do this. Please use the normal refinement block, as an additional iteration, or a new refinement block.  We will fix the filter block display.

**Reply by abartesaghi** [reply]
This issue is addressed in [v0.6.3](https://nextpyp.app/files/pyp/0.6.3/docs/index.html).

**Reply by JuhaHuiskonen** [reply]
I am now using the refinement block in v0.6.3.  The RELION output directory option is still ignored, but I can now see a `tomograms.star` file and `Movies/` being exported in the default directory `relion/`. However, there is no `particles.star` file. The job (Launch step) has been running now for 10 hours... Only these two lines are in the log:

```
1	2024-03-04 07:51:40 [INFO] /opt/pyp/bin/run/pyp:3232 | Job 20650532 (v0.6.3) launching on r18c29.bullx using 2 task(s) and 4 GB of RAM
2	2024-03-04 07:52:03 [INFO] pyp/inout/metadata/pyp_metadata.py:815 | Read pickle metadata, and update
```

**Reply by yezhou867** [reply]
If there are a lot of particles, it could take longer. But 10 hrs seems to be too long. Even if the 4G mem is not enough, there should be some error reported. Could you try a smaller dataset just to see if it works in your case?

**Reply by JuhaHuiskonen** [reply]
With 177000 particles it took about 15 hours.

**Reply by JuhaHuiskonen** [reply]
I am still struggling with this. Now I am trying to export the data from Tomography tutorial, after the first round of particle refinement. I created a new Particle refinement block (getting its inputs from the first), switch off refinements, and gave the parameter file in the  Export tab.

<img width="1405" alt="Screenshot 2024-03-17 at 14 49 45" src="https://github.com/nextpyp/.github/assets/20352352/6157ace4-cc0b-40bd-b8e0-06419cb91ed0">

Now the run finishes very quickly, but I get nothing written in the `relion` folder. 

The output log has one warning:

```
1	2024-03-17 11:00:08 [INFO] /opt/pyp/bin/run/pyp:3232 | Job 3306956 (v0.6.3) launching on c1303.mahti.csc.fi using 20 task(s) and 40 GB of RAM
2	2024-03-17 11:00:08 [INFO] /opt/pyp/bin/run/pyp:367 | Using parameter file /scratch/project_2009485/nextPYP/shared/users/huiskone/projects/tomo-tutorial-37VRb22R2avAHgjHP2/tomo-coarse-refinement-f1c58SOLD7zEvFkD/frealign/maps/tomo-coarse-refinement-f1c58SOLD7zEvFkD_r01_02.par.bz2
3	2024-03-17 11:00:08 [INFO] /opt/pyp/bin/run/pyp:371 | Using initial model /scratch/project_2009485/nextPYP/shared/users/huiskone/projects/tomo-tutorial-37VRb22R2avAHgjHP2/tomo-coarse-refinement-f1c58SOLD7zEvFkD/frealign/maps/tomo-coarse-refinement-f1c58SOLD7zEvFkD_r01_02.mrc
4	2024-03-17 11:00:08 [INFO] /opt/pyp/bin/run/pyp:375 | Using shape mask /scratch/project_2009485/nextPYP/shared/tutorial_data/tomo-tutorial/EMPIAR-10164_shape_mask.mrc
5	2024-03-17 11:00:09 [INFO] pyp/inout/metadata/pyp_metadata.py:815 | Read pickle metadata, and update
6	2024-03-17 11:23:21 [INFO] /opt/pyp/bin/run/pyp:3756 | PYP (export_star) finished successfully
7	/usr/local/envs/pyp/lib/python3.8/site-packages/numpy/core/fromnumeric.py:3464: RuntimeWarning: Mean of empty slice.
8	  return _methods._mean(a, axis=axis, dtype=dtype,
9	/usr/local/envs/pyp/lib/python3.8/site-packages/numpy/core/_methods.py:192: RuntimeWarning: invalid value encountered in scalar divide
10	  ret = ret.dtype.type(ret / rcount)
```
Edit: The export works, but the output folder name is still ignored.  Exported data is in folder `relion` under main folder of the current processing block.

**Reply by abartesaghi** [reply]
This was addressed in version [0.6.4](https://nextpyp.app/files/pyp/0.6.4/docs/index.html).
